compiler/rustc_typeck/src/check/expr.rs

```diff
@@ -1880,21 +1881,103 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         }
     }
 
-    fn no_such_field_err<T: Display>(
+    fn no_such_field_err(
         &self,
+        base_span: Span,
         span: Span,
-        field: T,
-        expr_t: &ty::TyS<'_>,
+        field: Ident,
+        expr_t: &'tcx ty::TyS<'tcx>,
     ) -> DiagnosticBuilder<'_> {
```

Comment:
It seems to me like you could just pass in `field` and use `field.span` instead of passing both `base_span` and `span`. The former seems unnecessary and the later is the same as `field.span`.